File: license-report/src/main/java/org/abhijitsarkar/licensereport/Group.java
package org.abhijitsarkar.licensereport;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

/**
 * @author Abhijit Sarkar
 */
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Group {
    private int id;
    private String name;
    @JsonProperty("web_url")
    private String url;
    private List<Project> projects;
}

File: license-report-kotlin/src/main/kotlin/org/abhijitsarkar/client/Group.kt
package org.abhijitsarkar.client;

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * @author Abhijit Sarkar
 */
class Group {
    lateinit var name: String
    var projects = emptyList<Project>()

    class Project {
        @JsonProperty("ssh_url_to_repo")
        lateinit var sshUrl: String
        lateinit var name: String
    }
}
